uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/CamelEditorTest.java

```diff
@@ -263,4 +265,36 @@ public void testDragAndDropComponents() {
 		assertTrue(EditorManipulator.isEditorContentEqualsFile("resources/camel-context-all.xml"));
 		assertTrue(LogGrapper.getPluginErrors("fuse").size() == 0);
 	}
+	
+	/**
+	 * <p>
+	 * Test checks collapse/expand feature of Camel editor components
+	 * </p>
+	 * <ol>
+	 * <li>create a new project from template 'Content Based Router'</li>
+	 * <li>open camel-context.xml file</li>
+	 * <li>Invoke 'Collapse' on route component 'Choice'</li>
+	 * <li>Check if component was collapsed (check if changed size of collapsed component area and height is smaller
+	 * then begin values)</li>
+	 * <li>Check fuse errors in Error Log View</li>
+	 * <li>Invoke 'Expand' on route component 'Choice'</li>
+	 * <li>Check if component was expanded (check changed size of expanded component area -> should be same as area size
+	 * and height before collapse)</li>
+	 * <li>Check fuse errors in Error Log View</li>
+	 * </ol>
+	 */
+	@Test
+	public void testCollapseExpandFeature() {
+		CamelEditor editor = new CamelEditor("camel-context.xml");
+		CamelComponentEditPart editPart = new CamelComponentEditPart("Choice");
+		int startHeight = editPart.getBounds().height;
+		editor.doOperation("Choice", "Collapse");
+		AbstractWait.sleep(TimePeriod.MEDIUM);
+		assertTrue("Route component 'Choice' wasn't collapsed properly", editPart.getBounds().height < startHeight);
+		assertTrue(LogGrapper.getPluginErrors("fuse").size() == 0);
+		editor.doOperation("Choice", "Expand");
+		AbstractWait.sleep(TimePeriod.MEDIUM);
+		assertEquals(editPart.getBounds().height, startHeight);
+		assertTrue(LogGrapper.getPluginErrors("fuse").size() == 0);
```

Comment:
it is checked at 33 places in the code.
For another PR, it might be useful to extract and provide some Util class to check for it.
Something like: LogChecker.assertNoFuseError()
![image](https://user-images.githubusercontent.com/1105127/33211488-0b66ccec-d11f-11e7-821c-f9a1810c0b94.png)